### PR TITLE
fix: Distinction between clearing data vs. layers

### LINKED ISSUE #1934: Emptying data from map settings deletes all layers
**Describe the bug**
When data gets emptied through the map settings menu, all layers are deleted as well.

**To Reproduce**
1. Go to any map (for example https://umap.openstreetmap.de/en/map/test-layer-selction-hidden-layer_61300)
2. Click on Edit
3. Click on Map Advanced Properties
4. open Advanced Actions and click Empty
5. All layers are getting completely erased

**Expected behavior**
Keep layers but delete all data (points/lines/areas) within them.

### PR DESCRIPTION
**Pull request by davidbgk**
Fix #1934

### CODE REVIEW

File: umap/static/umap/js/umap.js [hunk 1]
```diff
@@ -1505,8 +1505,15 @@ U.Map = L.Map.extend({
       L.DomUtil.createButton(
         'button umap-empty',
         advancedButtons,
-        L._('Empty'),
-        this.empty,
+        L._('Clear data'),
+        this.emptyDatalayers,
```
Comment (almet) [inline umap/static/umap/js/umap.js:1509]:
```suggestion
        this.emptyDataLayers,
```

File: umap/static/umap/js/umap.js [hunk 2]
```diff
@@ -1505,8 +1505,15 @@ U.Map = L.Map.extend({
...
+        this
+      )
+      L.DomUtil.createButton(
+        'button umap-empty',
+        advancedButtons,
+        L._('Remove layers'),
+        this.emptyMap,
```
Comment (almet) [inline umap/static/umap/js/umap.js:1516]: Maybe align and name it "removeLayer"? (or maybe it's already taken, and we should suffix these with "Action" or something similar ?)

File: umap/static/umap/js/umap.js [hunk 3]
```diff
@@ -1687,12 +1687,18 @@ U.Map = L.Map.extend({
...
   },
 
-  empty: function () {
+  emptyMap: function () {
     this.eachDataLayerReverse((datalayer) => {
       datalayer._delete()
     })
   },
 
+  emptyDatalayers: function () {
```
Comment (almet) [inline umap/static/umap/js/umap.js:1696]:
```suggestion
  emptyDataLayers: function () {
```

File: umap/static/umap/js/umap.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ L.Map.mergeOptions({
   // we cannot rely on this because of the y is overriden by Leaflet
   // See https://github.com/Leaflet/Leaflet/pull/9201
   // And let's remove this -y when this PR is merged and released.
-  demoTileInfos: { 's': 'a', 'z': 9, 'x': 265, 'y': 181, '-y': 181, 'r': '' },
+  demoTileInfos: { s: 'a', z: 9, x: 265, y: 181, '-y': 181, r: '' },
```
Comment (yohanboniface) [inline umap/static/umap/js/umap.js:16]: :upside_down_face:

**Review by yohanboniface — APPROVED** [review]

File: umap/static/umap/js/umap.js (hunk 4 above)
Comment (davidbgk) [reply umap/static/umap/js/umap.js:16]: I. WILL. WIN. 😅

**Pull request merged**